web/src/user_profile.ts

```diff
@@ -282,7 +282,8 @@ export function get_user_unsub_streams(user_id: number): dropdown_widget.Option[
 
 function format_user_stream_list_item_html(stream: StreamSubscription, user: User): string {
     const show_unsubscribe_button =
-        people.can_admin_user(user) || stream_data.can_unsubscribe_others(stream);
+        stream_data.can_unsubscribe_others(stream) ||
+        (people.is_my_user_id(user.user_id) && stream_data.can_unsubscribe(stream));
```

Comment:
API docs mention that user can unsubscribe their own bot if they have access to the channel irrespective of any seting. But this is not true here anymore.